token/program-2022/src/pod.rs

```diff
@@ -0,0 +1,186 @@
+//! Solana program utilities for Plain Old Data types
+use {
+    bytemuck::{Pod, Zeroable},
+    solana_program::{program_error::ProgramError, pubkey::Pubkey},
+};
+
+// TODO Generic `PodOption` is tricky, since we need to be sure that it'll
+// always take up the same amount of space.  A few ideas, since all of these are
+// pubkeys at the moment:
+//
+//#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+//#[repr(transparent)]
+//struct NonZeroPubkey(Pubkey);
+//impl NonZeroPubkey {
```

Comment:
It'll be a bit hacky to make it work, since the normal NonZero* int types get to use special libcore optimizations, but it's looking like the best option